### PR TITLE
chore: node_env as global env

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

This PR removes our dynamic implementation of Middlewares, as discussed on https://github.com/nodejs/nodejs.org/pull/5604, since it is a Next.js limitation to support dynamic matchers and to reduce the number of invocations on our Edge Middleware.

## Validation

This is validated by the Documentation of Next.js and after validating the indeed the amount of Edge Middleware invocations reduces.

### CODE REVIEW

**Comment by ovflowd** [comment]
@styfle do you have any idea why Windows builds are failing saying that "npm" is not a recognised command? Never seen this (https://github.com/nodejs/nodejs.org/actions/runs/5802085672/job/15727733041?pr=5613) could that be due to strict env?

File: middleware.ts [hunk 1]
```diff
@@ -1,9 +1,51 @@
 import { NextResponse } from 'next/server';
-import createMiddleware from './next.middleware';
-import detectLanguage from './middlewares/detectLanguage';
+import { availableLocales } from './next.locales.mjs';
+import type { NextRequest } from 'next/server';
 
-const nextMiddleware = createMiddleware(NextResponse);
+// This Middleware is responsible for handling automatic language detection from an user's Browser
```
Comment (styfle) [inline middleware.ts]:
```suggestion
// This Middleware is responsible for handling automatic language detection from a user's Browser
```

https://english.stackexchange.com/a/105117/9506

Comment (ovflowd) [reply middleware.ts]: Ty!

**Comment by ovflowd** [comment]
cc @nodejs/website I'm fast-tracking this as we're overusing a lot of Edge Middleware Invocations on our Vercel Cloud.

**Pull request merged**